src/main/java/com/adobe/aem/compgenerator/AemCompGenerator.java

```diff
@@ -40,44 +72,151 @@
     private static final Logger LOG = LogManager.getLogger(AemCompGenerator.class);
 
     public static void main(String[] args) {
-        try {
-            String configPath = "data-config.json";
-            if (args.length > 0) {
-                configPath = args[0];
-            }
+        // by default web app will run on port 8080
+        int port = 8080;
 
-            File configFile = new File(configPath);
+        if (args.length > 0) {
+            // optional ability to override default http port
+            String argument = args[0];
+            if (StringUtils.contains(argument, "p") &&
```

Comment:
are we looking or `-p` instead of just "p"?  Could this potentially match a wrong argument? (a config file path could easily have a "p" somewhere in it)

Reply:
fwiw I believe there are some good cli libraries out there for reading arguments and flags, if we think that would be helpful